src/fraction/approx.rs

```diff
@@ -441,38 +443,92 @@ impl<T: Clone + Integer + ToBigUint + ToBigInt + GenericInteger> GenericFraction
+        let fractional = (ratio.fract() * multiplier).to_integer();
 
-        assert_eq!(x, 5.into());
+        // Since `fractional` is an integer, converting it to a string removes any leading zeros.
+        // We're using it to represent the fractional part of the number, though, so we need to add
+        // back those leading zeros.
+        format!("{sign_str}{integer}.{fractional:0>decimal_places$}")
     }
 
-    #[test]
-    fn test_perf_10k() {
-        let _ = GenericFraction::<u8>::from(2_u8).sqrt_raw(10_000);
+    fn test_sqrt_of(value: impl std::borrow::Borrow<BigFraction>, simplify: bool, n: usize) {
+        let value = value.borrow();
+
+        let accuracy = Accuracy::decimal_places(n);
+
+        println!("calculating...");
+        let sqrt = value.sqrt_with_accuracy_raw(&accuracy);
+
+        let sqrt: BigFraction = if simplify {
+            {
+                println!("simplifying...");
+                sqrt.simplified()
+            }
+        } else {
+            sqrt
+        }
+        .into();
+
+        println!("checking...");
+        assert_eq!(
+            accuracy.chopped_numerator_raw(
+                &(sqrt.numer().unwrap() * sqrt.numer().unwrap()),
+                &(sqrt.denom().unwrap() * sqrt.denom().unwrap()),
+            ),
+            accuracy.chopped_numerator_raw(value.numer().unwrap(), value.denom().unwrap())
+        );
     }
 
-    #[test]
-    fn test_perf_100k() {
-        let _ = GenericFraction::<u8>::from(2_u8).sqrt_raw(100_000);
+    fn get_test_values() -> [BigFraction; 5] {
+        [
+            BigFraction::from(2),
+            BigFraction::from(123_654),
+            BigFraction::from(123_655),
+            BigFraction::from_str("5735874745115151552958367280658028638020529468164964850251033802750727314244020586751748892724760644/4789532131435371284839616979453671799246590610930954499621009334289181266216833845985099376094324166").unwrap(),
+            BigFraction::from(0.2),
+        ]
+    }
+
+    fn get_test_accuracies() -> [usize; 5] {
+        [10, 100, 1000, 10000, 100_000]
     }
 
     #[test]
-    fn test_big_numbers() {
-        let big_fraction = BigFraction::from_str("5735874745115151552958367280658028638020529468164964850251033802750727314244020586751748892724760644/4789532131435371284839616979453671799246590610930954499621009334289181266216833845985099376094324166").unwrap();
-        let sqrt = big_fraction.sqrt(1_000);
+    fn test_simplified() {
+        for value in get_test_values() {
+            for accuracy in get_test_accuracies() {
+                println!("sqrt({value}) to {accuracy} d.p., simplified");
+                test_sqrt_of(&value, true, accuracy);
+            }
+        }
+    }
 
-        let s = format!("{sqrt:.100}");
-        assert_eq!(s, "1.0943425455728974838600903859180783076888376493725431295813967125637781050743787384965051763360943812");
+    #[test]
+    fn test_raw() {
```

Comment:
I've split the tests into simplified and raw versions. Unfortunately, simplifying takes a very long time: `test_raw` completes in just under 12.5 seconds on my machine; `test_simplified` takes about 39 seconds. I can imagine that the former might be tolerable, but the latter seems excessive.

Do we need to test the simplified results, considering that it's just the same as `test_raw` but applying `num`'s simplifying routine on each value? My instinct is to test everything, but here I think maybe it would be fine not to?

Reply:
Agree, if we rely on the external `simplify` functionality, we should be fine to assume it works as expected and only test the raw versions.
On the other hand, I don't think 39 seconds is a totally unaccepted. I think we should be fine to take it as long as it's below 1 minute to run (for one module).

Reply:
Great, thanks. I'll get rid of `test_simplified` and perhaps extend `test_raw` to be more comprehensive, given that we've got a bit more time to spare.